src/logging/LoggingService.ts

```diff
@@ -14,77 +14,38 @@
  * limitations under the License.
  */
 
+import {Property} from '../config/Properties';
 import {DefaultLogger} from './DefaultLogger';
-import {NoLogger} from './NoLogger';
+import {ILogger} from './ILogger';
 
 export enum LogLevel {
+    OFF = -1,
     ERROR = 0,
     WARN = 1,
     INFO = 2,
     DEBUG = 3,
     TRACE = 4,
 }
 
-export interface ILogger {
-    log(level: LogLevel, className: string, message: string, furtherInfo: any): void;
-}
-
 export class LoggingService {
 
-    private static loggingService: LoggingService;
-    logger: ILogger;
-
-    constructor(externalLogger: ILogger = null) {
-        if (externalLogger != null) {
-            this.logger = externalLogger;
-        }
-        if (this.logger == null) {
-            this.logger = new DefaultLogger();
-        }
-    }
-
-    static getLoggingService(): LoggingService {
-        if (LoggingService.loggingService == null) {
-            LoggingService.initialize(null);
-        }
-        return LoggingService.loggingService;
-    }
+    private readonly logger: ILogger;
 
-    static initialize(loggerModule: string | ILogger = null): void {
-        if (typeof loggerModule === 'string') {
-            if (loggerModule === 'off') {
-                LoggingService.loggingService = new LoggingService(new NoLogger());
-            } else if (loggerModule === 'default') {
-                LoggingService.loggingService = new LoggingService();
-            } else {
-                throw new RangeError('Logging type unknown: ' + loggerModule);
-            }
+    constructor(loggingProperty: Property, logLevel: number) {
```

Comment:
I see the option of completely removing the logger is done via `-1` logging level now. With this method, everytime a log call is made, it's log level will be compared to `-1` and the log function will not do anything because the comparison is false. However, we do an unneccessary comparison at every log call. The purpose of `NoLogger` was to suppress this unneccessary comparison. I think we should not remove `NoLogger`. We can check the configured log level once the LoggingService is first initialized and we can start a NoLogger in case of -1.

Reply:
You are right about that it makes extra comparison but I think it is not a huge thing. Also, it provides consistency and simplicity. Therefore, I am in the favor of removing `NoLogger`.

Reply:
It is an optimization. It is this PR making the change. Therefore, the PR needs justification not the current state of the code.

Reply:
In the first implementation, it wasn't a log level. It was a logger like default. However, now, it is a log level and there is no difference from others. I think it can justify why it is removed.